walkeeper/src/handler.rs

```diff
@@ -53,51 +91,52 @@ impl postgres_backend::Handler for SafekeeperPostgresHandler {
         Ok(())
     }
 
-    fn process_query(&mut self, pgb: &mut PostgresBackend, query_string: Bytes) -> Result<()> {
-        // START_WAL_PUSH is the only command that initializes the timeline in production.
-        // There is also JSON_CTRL command, which should initialize the timeline for testing.
-        if self.timeline.is_none() {
-            if query_string.starts_with(b"START_WAL_PUSH") || query_string.starts_with(b"JSON_CTRL")
-            {
-                self.timeline.set(
-                    &self.conf,
-                    self.tenantid.unwrap(),
-                    self.timelineid.unwrap(),
-                    CreateControlFile::True,
-                )?;
-            } else {
-                self.timeline.set(
-                    &self.conf,
-                    self.tenantid.unwrap(),
-                    self.timelineid.unwrap(),
-                    CreateControlFile::False,
-                )?;
+    fn process_query(&mut self, pgb: &mut PostgresBackend, query_string: &str) -> Result<()> {
+        let cmd = parse_cmd(query_string)?;
+
+        // Is this command is ztimeline scoped?
+        match cmd {
+            SafekeeperPostgresCommand::StartWalPush { .. }
+            | SafekeeperPostgresCommand::StartReplication { .. }
+            | SafekeeperPostgresCommand::IdentifySystem
+            | SafekeeperPostgresCommand::JSONCtrl { .. } => {
+                let tenantid = self
+                    .ztenantid
+                    .ok_or_else(|| anyhow!("tenantid is required"))?;
+                let timelineid = self
+                    .ztimelineid
+                    .ok_or_else(|| anyhow!("timelineid is required"))?;
```

Comment:
So, now ztenatntid and ztimelineid are required for all 4 commands, right?
It seems that this code can be simplified.

Reply:
Yes, but I'm going to change this soon.